---
title: "Introduction to R function _assessPOE_"
author: "Yuang Tian"
date: "2021/3/8"
output: pdf_document
vignette: >
  %\VignetteIndexEntry{Introduction to CCMO}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

The R function _assessPOE_ implements an efficient statistical method for assessing parent-of-origin effects (POEs) using case-control mother-offspring data.

## 1. Data description

The data for each mother-offspring pair are $(Y,G^c,G^m,X)$, where $Y$ is the disease status of the offspring (should be coded as 1 for case and 0 for control), $G^c$ and $G^m$ are the genotype vectors at the target SNP and adjacent SNPs for the mother and child, and $X$ is a $p$-vector of maternal covariates. 

Suppose a SNP has two alleles A and a (a is the minor allele), then genotypes can be coded either as 0 (for genotype AA), 1 (for genotype Aa), or 2 (for genotype aa).

## 2. Model description

The penetrance model for parent-of-origin effects is as follows:
\begin{eqnarray*}
&&\mbox{pr}(Y=1|g^m,g^c,g^c_m,g^c_p,X)\\
&=&\mbox{expit}(\beta_0+\beta_{g^m}g^m+\beta_{g^c}g^c+\beta_{im}(g^c_m-g^c_p)+\beta_{X}X),
\end{eqnarray*}
where $\mbox{expit}(\cdot) = e^\cdot/(1 + e^\cdot)$ is the expit function.  Here $(g^m,g^c)$ denote the target SNP genotypes for a mother-child pair, $g^c_m$ and $g^c_p$ denote children alleles at the target SNP inherited from the mother and father, respectively, coded as 0 and 1 for A and a, respectively. $\beta_{g^c}$ and $\beta_{g^m}$ are main genetic effects for offspring (child) and mother, respectively, $\beta_X$ is the main maternal environmental effect, $\beta_{im}$ is the POE. We also considered the version with interaction terms:
\begin{eqnarray*}
&&\mbox{pr}(Y=1|g^m,g^c,g^c_m,g^c_p,X)\\
&=&\mbox{expit}(\beta_0+\beta_{g^m}g^m+\beta_{g^c}g^c+\beta_{im}(g^c_m-g^c_p)+\beta_{X}X+\beta_{g^mX}g^mX+\beta_{g^cX}g^cX),
\end{eqnarray*} where $\beta_{g^mX}$ and $\beta_{g^cX}$ denote the gene-environment interaction effects. In additional to multi-locus genotypes, disease statuses, and covariates, the inputs for the POE analysis function _assessPOE_ should include possible haplotypes in the population of interest, which can be generated by the function _est.haplo_. 

## 3. Usage of main function _assessPOE_

The usage of the function _assessPOE_ is as follows:

```{r,eval=FALSE}
assessPOE(Y,gmm,gcc,X,test.locus,hap,f,method,inter)
```



Users should specify test locus and prevalence _f_, and make sure that the genotypes are coded as 0 (for genotype AA), 1 (for genotype Aa), or 2 (for genotype aa). Users should specify _inter_ to choose the model they prefer to use ( _TRUE_ for model with interaction effects and _FALSE_ for model without interaction effects), and specify _method_ they prefer to use from _'ROB-HAP'_, _'IND-HAP'_, _'ROB-HAP-EM'_, and _'IND-HAP-EM'_ (more details refer to the reference). 

Users can use _est.haplo_ to generate inputs _hap_ for _assessPOE_. 
```{r,eval=FALSE}
est.haplo(gmm)
```

Note that the input disease status _Y_ must be a numeric vector, genotypes _gmm_ and _gcc_ must be numeric matrices, and maternal coviates _X_ can be a numeric vector or a numeric matrix.

## 4. Illustration of main function _assessPOE_
```{r,eval=FALSE}
load(POESampleData.rda)
Y <- POESampleData[,1]
gmm <- POESampleData[,2:6]
gcc <- POESampleData[,7:11]
X <- POESampleData[,12]
hap <- est.haplo(gmm)
test.locus <- 1
f <- 0.01
fit <- assessPOE(Y,gmm,gcc,X,test.locus,hap,f,'ROB-HAP',FALSE)
```

The point estimation results are stored in
```{r,eval=FALSE}
fit$est
```
The estimated standard errors of the estimates are stored in
```{r,eval=FALSE}
fit$sd
```
The corresponding results by the standard logistic regression are included for comparison
```{r,eval=FALSE}
fit$est.log
fit$sd.log
```

## 5. Reference

Tian Y, Zhang H, Bureau A, Hochner H, Chen J. (2021). An Efficient Method for Assessing Parent-of-Origin Effects Using Case-Control Mother-Offspring Pairs. Manuscript.
